src/google/protobuf/compiler/csharp/csharp_message.cc

```diff
@@ -61,20 +61,27 @@ bool CompareFieldNumbers(const FieldDescriptor* d1, const FieldDescriptor* d2) {
 MessageGenerator::MessageGenerator(const Descriptor* descriptor,
                                    const Options* options)
     : SourceGeneratorBase(descriptor->file(), options),
-      descriptor_(descriptor) {
-
-  // sorted field names
-  for (int i = 0; i < descriptor_->field_count(); i++) {
```

Comment:
If this can be removed, the field_nams_ definition in charp_message.h and the "const std::vector<std::string>& MessageGenerator::field_names() " below should also be removed.